api/v1alpha08/sonataflow_types.go

```diff
@@ -24,6 +24,535 @@ import (
+	// will be made available to those containers which consume them
+	// by name.
+	//
+	// This is an alpha field and requires enabling the
+	// DynamicResourceAllocation feature gate.
+	//
+	// This field is immutable.
+	//
+	// +patchMergeKey=name
+	// +patchStrategy=merge,retainKeys
+	// +listType=map
+	// +listMapKey=name
+	// +featureGate=DynamicResourceAllocation
+	// +optional
+	ResourceClaims []corev1.PodResourceClaim `json:"resourceClaims,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"name" protobuf:"bytes,39,rep,name=resourceClaims"`
+}
+
+func (f *FlowPodSpec) ToPodSpec() corev1.PodSpec {
+	return corev1.PodSpec{
+		Volumes:                       f.Volumes,
+		InitContainers:                f.InitContainers,
+		Containers:                    f.Containers,
+		RestartPolicy:                 f.RestartPolicy,
+		TerminationGracePeriodSeconds: f.TerminationGracePeriodSeconds,
+		ActiveDeadlineSeconds:         f.ActiveDeadlineSeconds,
+		DNSPolicy:                     f.DNSPolicy,
+		NodeSelector:                  f.NodeSelector,
+		ServiceAccountName:            f.ServiceAccountName,
+		AutomountServiceAccountToken:  f.AutomountServiceAccountToken,
+		NodeName:                      f.NodeName,
+		HostNetwork:                   f.HostNetwork,
+		HostPID:                       f.HostPID,
+		HostIPC:                       f.HostIPC,
+		ShareProcessNamespace:         f.ShareProcessNamespace,
+		SecurityContext:               f.SecurityContext,
+		ImagePullSecrets:              f.ImagePullSecrets,
+		Hostname:                      f.Hostname,
+		Subdomain:                     f.Subdomain,
+		Affinity:                      f.Affinity,
+		SchedulerName:                 f.SchedulerName,
+		Tolerations:                   f.Tolerations,
+		HostAliases:                   f.HostAliases,
+		PriorityClassName:             f.PriorityClassName,
+		Priority:                      f.Priority,
+		DNSConfig:                     f.DNSConfig,
+		ReadinessGates:                f.ReadinessGates,
+		RuntimeClassName:              f.RuntimeClassName,
+		EnableServiceLinks:            f.EnableServiceLinks,
+		PreemptionPolicy:              f.PreemptionPolicy,
+		Overhead:                      f.Overhead,
+		TopologySpreadConstraints:     f.TopologySpreadConstraints,
+		SetHostnameAsFQDN:             f.SetHostnameAsFQDN,
+		OS:                            f.OS,
+		HostUsers:                     f.HostUsers,
+		SchedulingGates:               f.SchedulingGates,
+		ResourceClaims:                f.ResourceClaims,
+	}
+}
+
+// FlowPodSpecTemplate describes the desired custom Kubernetes PodSpecTemplate definition for the deployed flow.
+//
+// The FlowContainer describes the container where the actual flow is running. It will override any default definitions.
+// For example, to override the image one can use `.spec.podSpecTemplate.flowContainer.image = my/image:tag`.
+type FlowPodSpecTemplate struct {
```

Comment:
Doing an analogy with the Kubernetes API, where we have corev1.PodTemplateSpec, I think this type could be named FlowPodTemplateSpec instead.